lean4-info.el

```diff
@@ -82,19 +82,25 @@
 (lsp-defun lean4-diagnostic-full-end-line ((&lean:Diagnostic :full-range (&Range :end (&Position :line))))
   line)
 
-(defun lean4-mk-message-section (caption errors)
+(defun lean4-mk-message-section (uri caption errors)
   (when errors
     (magit-insert-section (magit-section)
       (magit-insert-heading caption)
       (magit-insert-section-body
         (dolist (e errors)
           (-let (((&Diagnostic :message :range (&Range :start (&Position :line :character))) e))
-            (magit-insert-section (magit-section)
-              (magit-insert-heading (format "%d:%d: " (1+ (lsp-translate-line line)) (lsp-translate-column character)))
+	    (magit-insert-section (magit-section)
+	      (magit-insert-heading)
+	      (insert-button (format "%s %d:%d" uri (1+ (lsp-translate-line line)) (lsp-translate-column character))
+			     'action (lambda (btn)
+				       (with-current-buffer (find-file-other-window (lsp--uri-to-path uri))
```

Comment:
I imagine `emacs` has a nicer API to open the file and go to the correct file, line and column?